pkg/resources/common/common.go

```diff
@@ -14,3 +22,79 @@ func ConvertAuthnExtras(extra map[string]authnv1.ExtraValue) map[string]authzv1.
 	}
 	return result
 }
+
+// ValidateLabel checks if a user is removing or modifying a label. If the label is newly added, return false.
+func IsModifyingLabel(oldLabels, newLabels map[string]string, label string) bool {
+	var oldValue, newValue string
+	var oldLabelExists, newLabelExists bool
+	if oldLabels == nil {
+		oldLabelExists = false
+	} else {
+		oldValue, oldLabelExists = oldLabels[label]
+	}
+	if newLabels == nil {
+		newLabelExists = false
+	} else {
+		newValue, newLabelExists = newLabels[label]
+	}
+
+	if !oldLabelExists {
+		return false
+	}
+	if oldLabelExists && !newLabelExists {
+		return true
+	}
+	if oldValue != newValue {
+		return true
+	}
+
+	return false
+}
+
+// cachedVerbChecker is used for caching if a request for a non-namespaced gvr with specified name has the given overrideVerb.
+// This is meant to eliminate the need to perform multiple calls to the provided SubjectAccessReview for the overrideVerb.
+type cachedVerbChecker struct {
```

Comment:
A couple of overall comments on this struct:
- From what I've heard, typically in go you want to avoid constructors and allow objects to be constructed directly. @KevinJoiner might feel more strongly than I do on that point, and may have an opinion on if this is permissible in this case.
- It's a bit odd to have an unexported type, which doesn't implement an interface, returned from an exported function. I think your CI is failing because of this behavior (and an associated revive check) so you will probably need to adjust this
- I'd recommend emphasizing in your comment the limitations of this struct. Namely:
  - Each admitter must request a new verbChecker for each request.
  - Verb checkers must not be shared across different admitters.
  - If the caller needs to change what it is checking (regardless of if it's a different verb, resource name, or resourceType), then they need to make a separate verb checker.
 The reason I'm emphasizing these limitations is that this struct is used for access control, so a bug here is likely to turn into a CVE. We need to be very clear here to avoid issues.